### ISSUE TITLE
Fix CVE-2021-24891

### ISSUE THREAD

**Issue by karkis3c** [projectdiscovery/nuclei-templates#10642]
[CVE-2021-24891.yaml](https://github.com/projectdiscovery/nuclei-templates/blob/main/http/cves/2021/CVE-2021-24891.yaml) isn't working - template is invalid, the detected version doesn't work, and the exploit is missing.

**Working Template link:**

https://raw.githubusercontent.com/karkis3c/bugbounty/main/nuclei-templates/cves/CVE-2021-24891.yaml

Thanks!

**Comment by karkis3c** [comment]
id: CVE-2021-24891

info:
  name: WordPress Elementor Website Builder < 3.4.8 - Cross-Site Scripting
  author: karkis3c
  severity: medium
  description: |
    WordPress Elementor Website Builder plugin before 3.4.8 contains a DOM cross-site scripting vulnerability.
  poc: |
    https://wpscan.com/vulnerability/fbed0daa-007d-4f91-8d87-4bca7781de2d/
  reference:
    - https://www.jbelamor.com/xss-elementor-lightox.html
    - https://nvd.nist.gov/vuln/detail/CVE-2021-24891
    - https://github.com/karkis3c/bugbounty/blob/main/nuclei-templates/cves/CVE-2021-24891.yaml

flow: http(1) && http(2)

http:
  - method: GET
    path:
      - "{{BaseURL}}/wp-content/plugins/elementor/readme.txt"

    matchers-condition: and
    matchers:
      - type: word
        internal: true
        words:
          - "Stable tag:"
        part: body

    extractors:
      - type: regex
        name: version
        part: body
        group: 1
        internal: true
        regex:
          - "Stable tag: ([0-9]+\\.[0-9]+\\.[0-9]+)"

  - method: GET
    path:
      - "{{BaseURL}}/wp-content/plugins/elementor/readme.txt"

    matchers-condition: and
    matchers:
      - type: dsl
        dsl:
          - compare_versions(version, "<3.4.8")

    extractors:
      - type: dsl
        dsl:
          - 'version'

**Comment by DhiyaneshGeek** [comment]
Hi @karkis3c 

can you confirm the below template works for you

```
id: CVE-2021-24891

info:
  name: WordPress Elementor Website Builder <3.1.4 - Cross-Site Scripting
  author: dhiyaneshDk
  severity: medium
  description: |
    WordPress Elementor Website Builder plugin before 3.1.4 contains a DOM cross-site scripting vulnerability. It does not sanitize or escape user input appended to the DOM via a malicious hash.
  impact: |
    Successful exploitation of this vulnerability could allow an attacker to execute arbitrary JavaScript code in the context of the victim's browser, leading to potential data theft, session hijacking, or defacement of the affected website.
  remediation: |
    Update WordPress Elementor Website Builder to version 3.1.4 or later to mitigate this vulnerability.
  reference:
    - https://www.jbelamor.com/xss-elementor-lightox.html
    - https://wpscan.com/vulnerability/fbed0daa-007d-4f91-8d87-4bca7781de2d
    - https://nvd.nist.gov/vuln/detail/CVE-2021-24891
    - https://github.com/ARPSyndicate/kenzer-templates
  classification:
    cvss-metrics: CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N
    cvss-score: 6.1
    cve-id: CVE-2021-24891
    cwe-id: CWE-79
    epss-score: 0.00116
    epss-percentile: 0.45236
    cpe: cpe:2.3:a:elementor:website_builder:*:*:*:*:*:wordpress:*:*
  metadata:
    max-request: 3
    vendor: elementor
    product: "website_builder"
    framework: wordpress
  tags: cve2021,cve,wordpress,wp-plugin,elementor,wpscan,dom,xss
flow: http(1) && http(2)

http:
  - raw:
      - |
        GET /wp-content/plugins/elementor/readme.txt HTTP/1.1
        Host: {{Hostname}}

    matchers:
      - type: word
        words:
          - 'Elementor Website Builder'
          - 'Elementor Page Builder'
        internal: true

  - method: GET
    path:
      - "{{BaseURL}}/wp-content/plugins/elementor/assets/js/frontend.min.js"

    matchers-condition: and
    matchers:
      - type: dsl
        dsl:
          - compare_versions(version, '> 1.5.0', '< 3.1.4')

      - type: regex
        part: body
        regex:
          - "elementor[\\s-]*v(([0-3]+\\.(([0-5]+\\.[0-5]+)|[0-4]+\\.[0-9]+))|[0-2]+[0-9.]+)"

    extractors:
      - type: regex
        name: version
        group: 1
        regex:
          - "elementor[\\s-]*v(([0-3]+\\.(([0-5]+\\.[0-5]+)|[0-4]+\\.[0-9]+))|[0-2]+[0-9.]+)"
        internal: true

      - type: kval
        kval:
          - version
# digest: 490a0046304402202623889f6a04d4102a3ee41ca9948d2142f9f84fa0da1e8ef8ffcd628e93656102202e85312c8ea957f1764523d65a51f68b8df842f32c4367f338da6dd2c1fe2a36:922c64590222798bb761d5b6d8e72950
```

**Comment by DhiyaneshGeek** [comment]
Hi @karkis3c 

as discussed , i have fixed the template and raised PR #10648 

Thanks a lot for brining to our attention 😄
